FirebasePerformance/Sources/FPRClient.m

```diff
@@ -237,7 +237,10 @@ - (void)logGaugeMetric:(nonnull NSArray *)gaugeData forSessionId:(nonnull NSStri
   }
   dispatch_group_async(self.eventsQueueGroup, self.eventsQueue, ^{
     FPRMSGPerfMetric *metric = FPRGetPerfMetricMessage(self.config.appID);
-    FPRMSGGaugeMetric *gaugeMetric = FPRGetGaugeMetric(gaugeData, sessionId);
+    FPRMSGGaugeMetric *gaugeMetric = nil;
+    if ((gaugeData != nil && gaugeData.count != 0) && (sessionId != nil && sessionId.length != 0)) {
```

Comment:
Why would we have to check this here? Also, thinking wild if we can do this check in the GaugeMetric data structure for completeness?

Reply:
The check was  in https://github.com/firebase/firebase-ios-sdk/blob/master/FirebasePerformance/Sources/FPRProtoUtils.m#L198-L204. Because `GetGaugeMetric` cannot return nil, I moved the checks here. I think it makes sense to have the check in FPRClient because it was passed by FPRClient and is not a part of a GaugeMetric object.

Reply:
Sounds good, makes sense.